### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,44 @@
...
+      case 'removeProperties':
+        removeProperties(action.keysToRemove, stateCopy);
+        break;
+
+      case 'clear':
+        clear(stateCopy);
+        break;
+
+      default:
+        throw new Error('Error');
```
Comment (ruslanpashkov) [inline src/transformStateWithClones.js:29]: it would be better to specify which error we will get

**Review by ruslanpashkov — APPROVED** [review]

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,44 @@
...
+    }
+
+    stateVersions.push({ ...stateCopy });
+  }
+
+  return stateVersions;
+}
+
+function removeProperties(keysToRemove, state) {
```
Comment (silchyn) [inline src/transformStateWithClones.js:38]: what about `addProperties`?

**Review by silchyn — CHANGES_REQUESTED** [review]
Pay attention to comment above

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,50 @@
...
+  return stateVersions;
+}
+
+function addProperties(state, propertiesToAdd) {
+  for (const property in propertiesToAdd) {
+    state[property] = propertiesToAdd[property];
+  }
+}
+
+function removeProperties(keysToRemove, state) {
```
Comment (33Orange) [inline src/transformStateWithClones.js:44]: The order of parameters in the `removeProperties` function definition should be swapped. It should be `function removeProperties(state, keysToRemove) {`

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,50 @@
...
+  const stateVersions = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(action.keysToRemove, stateCopy);
```
Comment (33Orange) [inline src/transformStateWithClones.js:20]: The order of arguments in the `removeProperties` function should be swapped. It should be `removeProperties(stateCopy, action.keysToRemove);`

File: src/transformStateWithClones.js [hunk 5]
```diff
@@ -7,7 +7,50 @@
...
+    stateVersions.push({ ...stateCopy });
+  }
+
+  return stateVersions;
+}
+
+function addProperties(state, propertiesToAdd) {
+  for (const property in propertiesToAdd) {
+    state[property] = propertiesToAdd[property];
+  }
```
Comment (33Orange) [inline src/transformStateWithClones.js:41]: Let's change this part of the code to `Object.assign`

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job !

A few moments:
- Let's change addProperties function to code with `Object.assign` (Checklist p1).
- Some points about readability

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**